lib/inspec/resources/mail_alias.rb

```diff
@@ -0,0 +1,46 @@
+require "inspec/resources/command"
+module Inspec::Resources
+  class Mailalias < Inspec.resource(1)
+    # resource internal name.
+    name "mail_alias"
+
+    # Restrict to only run on the below platforms (if none were given,
+    # all OS's and cloud API's supported)
+    supports platform: "unix"
+
+    desc "Use the mail_alias InSpec audit resource to test mail alias present in the aliases file"
+
+    example <<~EXAMPLE
+      describe mail_alias("toor") do
+        it { should be_aliased_to "root" }
+      end
+    EXAMPLE
+
+    def initialize(alias_key)
+      skip_resource "The `mail_alias` resource is not yet available on your OS." unless inspec.os.unix?
+      @alias_key = alias_key
+    end
+
+    # resource_id is used in reporting engines to uniquely identify the individual resource.
+    def resource_id
+      "#{@alias_key}"
+    end
+
+    # resource appearance in test reports.
+    def to_s
+      "mail_alias #{resource_id}"
+    end
+
+    # aliased_to matcher checks if the given alias_value is set to the initialized alias_key
+    def aliased_to?(alias_value)
+      # /etc/aliases if the file where the alias and its value(s) are stored
+      cmd = inspec.command("cat /etc/aliases | grep #{@alias_key}")
```

Comment:
`cmd = inspec.command("cat /etc/aliases | grep '^#{@alias_key}:'")`
Does this look better the grep matches for sentence starting with the alias followed by colon?

Reply:
I think you will need egrep, not just grep.

Reply:
The regex meta characters which are required to be escaped while using grep are `'{', '}', '(', ')', '|', '+', '?'`, since none of them are used as part of `'^#{@alias_key}:'`, I think grep and egrep would work the same in this situation. If we would like to be more strict with the regex we could still go with `grep -e` option instead of `egrep` since _Direct invocation as either egrep or fgrep is deprecated_  Source: [gnu.org](https://www.gnu.org/software/grep/manual/grep.html#:~:text=Direct%20invocation%20as%20either%20egrep,on%20them%20to%20run%20unmodified.)